entity-framework/core/miscellaneous/nullable-reference-types.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Nullable reference types - EF Core
+author: roji
+ms.date: 9/9/2019
+ms.assetid: bde4e0ee-fba3-4813-a849-27049323d301
+uid: core/miscellaneous/nullable-reference-types
+---
+# Nullable Reference Types
+
+C# 8 introduced a new feature called [nullable reference types](/dotnet/csharp/tutorials/nullable-reference-types), allowing reference types to be annotated, indicating whether it is valid for them to contain null or not. If you are new to this feature, it is recommended that make yourself familiar with it by reading the C# docs.
+
+This page introduces EF Core's support fo nullable reference types, and describes best practices for working with them.
+
+## Required and optional properties
+
+If nullable reference types are enabled, EF Core will by convention configure reference properties as required or optional based on their C# type nullability. For example, properties with type `string` will be configured as required, while properties with type `string?` will be configured as optional.
+
+[!code-csharp[Main](../../../samples/core/Miscellaneous/NullableReferenceTypes/Customer.cs?highlight=7-9)]
+
+Using nullable reference types is recommended since it flows the nullability expressed in C# code to EF Core's model and to the database, and obviates the use of the Fluent API or Data Annotations to express the same concept twice.
+
+> [!NOTE]
+> Exercise caution when enabling nullable reference types on an existing project: reference type properties which were previously configured as optional will now be configured as required, unless they are explicitly annotated to be nullable. When managing a relational database schema, this may cause migrations to be generated which alter the database column's nullability.
+
+## DbContext and DbSet
```

Comment:
I find that page a bit confusing and not necessarily suitable for newcomers (e.g. it starts with "Design-time DbContext configuration" which seems very specific and somewhat advanced). I think core/modeling/index could be a better candidate. But see also my comments above on a specific nullability page etc.

Reply:
Yes, I agree the configuring DbContext page can be improved, reorganized, and that some of the more basic/introductory content is currently missing from it. However it exists for good reasons and many of the more advanced things including in it are there because they happen to be too common pitfalls that we have been forced to document.

If the root modeling page is a better candidate to explain how the DbSets on the DbContext contribute to the model, I guess the configuring DbContext page should have a paragraph and a link to that page too.